// test protos

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.unitbasicenv";
option java_outer_classname = "UnityBasicEnvProto";
option objc_class_prefix = "UBE";

package unitybasicenv;

// Interface exported by the server.
service UnityBasicEnv {
 
  // A Bidirectional streaming RPC.
  //
  // Accepts a stream of UnityBasicEnvNote sent while a route is being traversed,
  // while receiving other UnityBasicEnvNote (e.g. from other users).
  rpc UnityBasicEnvChat(stream UnityBasicEnvNote) returns (stream UnityBasicEnvNote) {}
}

// A UnityBasicEnvNote is a message sent while at a given point.
message UnityBasicEnvNote {
  // The message to be sent.
  string message = 1;
  //messages from server: step, reset
  //messages from client: start, connect, settings, action

  //when coming from client to server its an action
  //when coming from server to client its an obs (spot where the unit is)
  int32 actionObs = 2;
  //reward from the env (generated by server)
  float reward = 3;
  //true if env episode is done (generated by server)
  bool done = 4;
}
